[Task]: Update GoogleDrive VFS

### What needs to happen?

The GoogleDrive VFS uses old style key and some of the underlying libraries are outdated.
Also make it possible to use the instance role to use authentication.

### Issue Priority

Priority: 2

### Issue Component

Component: VFS

Same problem here.

Config:

![image](https://user-images.githubusercontent.com/70301866/224782384-aa6db4c1-e9b6-40a8-9ec7-2313f7902980.png)

Json:

![image](https://user-images.githubusercontent.com/70301866/224782562-2295fcdb-88a9-4d98-9502-c749e319e04b.png)

Logs Text Input:

Error browsing to location: 'googledrive://'
HopFileException: 

Unable to get VFS File object for filename 'googledrive://' : Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".
Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".

Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".

Root cause: IllegalArgumentException: 

org.apache.hop.core.exception.HopFileException: 

Unable to get VFS File object for filename 'googledrive://' : Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".
Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".

Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".

	at org.apache.hop.core.vfs.HopVfs.getFileObject(HopVfs.java:194)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.refreshBrowser(HopVfsFileDialog.java:878)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.enteredFilenameOrFolder(HopVfsFileDialog.java:684)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.lambda$open$2(HopVfsFileDialog.java:329)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4251)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4068)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3645)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.open(HopVfsFileDialog.java:581)
	at org.apache.hop.ui.core.dialog.BaseDialog.presentFileDialog(BaseDialog.java:263)
	at org.apache.hop.ui.core.dialog.BaseDialog.presentFileDialog(BaseDialog.java:163)
	at org.apache.hop.pipeline.transforms.fileinput.text.TextFileInputDialog.lambda$open$8(TextFileInputDialog.java:470)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4251)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4068)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3645)
	at org.apache.hop.ui.core.dialog.BaseDialog.defaultShellHandling(BaseDialog.java:513)
	at org.apache.hop.pipeline.transforms.fileinput.text.TextFileInputDialog.open(TextFileInputDialog.java:487)
	at org.apache.hop.ui.hopgui.file.pipeline.delegates.HopGuiPipelineTransformDelegate.editTransform(HopGuiPipelineTransformDelegate.java:174)
	at org.apache.hop.ui.hopgui.file.pipeline.HopGuiPipelineGraph.editTransform(HopGuiPipelineGraph.java:3342)
	at org.apache.hop.ui.hopgui.file.pipeline.HopGuiPipelineGraph.editTransform(HopGuiPipelineGraph.java:2163)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.hop.core.gui.plugin.action.GuiActionLambdaBuilder.lambda$createLambda$0(GuiActionLambdaBuilder.java:96)
	at org.apache.hop.ui.hopgui.context.GuiContextUtil.lambda$handleActionSelection$0(GuiContextUtil.java:183)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4043)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3648)
	at org.apache.hop.ui.hopgui.HopGui.open(HopGui.java:464)
	at org.apache.hop.ui.hopgui.HopGui.main(HopGui.java:351)
Caused by: org.apache.commons.vfs2.FileSystemException: Unknown message with code "Unable to get Google Drive file object for 'googledrive:///'".
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.processFile(GoogleDriveFileSystem.java:75)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.resolveFile(GoogleDriveFileSystem.java:54)
	at org.apache.commons.vfs2.provider.AbstractOriginatingFileProvider.findFile(AbstractOriginatingFileProvider.java:75)
	at org.apache.commons.vfs2.provider.AbstractOriginatingFileProvider.findFile(AbstractOriginatingFileProvider.java:55)
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveFile(DefaultFileSystemManager.java:788)
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveFile(DefaultFileSystemManager.java:754)
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveFile(DefaultFileSystemManager.java:819)
	at org.apache.hop.core.vfs.HopVfs.getFileObject(HopVfs.java:188)
	... 34 more
Caused by: org.apache.commons.vfs2.FileSystemException
	at org.apache.hop.vfs.googledrive.GoogleDriveFileObject.<init>(GoogleDriveFileObject.java:99)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.createFile(GoogleDriveFileSystem.java:40)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.processFile(GoogleDriveFileSystem.java:73)
	... 41 more
Caused by: java.lang.IllegalArgumentException
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:131)
	at com.google.api.client.util.Preconditions.checkArgument(Preconditions.java:35)
	at com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets.getDetails(GoogleClientSecrets.java:80)
	at com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow$Builder.<init>(GoogleAuthorizationCodeFlow.java:211)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileObject.getCredentials(GoogleDriveFileObject.java:245)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileObject.getDriveService(GoogleDriveFileObject.java:255)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileObject.<init>(GoogleDriveFileObject.java:96)
	... 43 more

Ow.. new error.


the consent screen appeared to give permission and then it displayed the error below

Error browsing to location: 'googledrive://'
ExceptionInInitializerError: 
Root cause: IllegalStateException: You are currently running with version 2.0.1 of google-api-client. You need at least version 1.15 of google-api-client to run version 1.25.0 of the Drive API library.

java.lang.ExceptionInInitializerError
	at com.google.api.services.drive.Drive$Builder.build(Drive.java:10116)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileObject.getDriveService(GoogleDriveFileObject.java:258)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileObject.<init>(GoogleDriveFileObject.java:96)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.createFile(GoogleDriveFileSystem.java:40)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.processFile(GoogleDriveFileSystem.java:73)
	at org.apache.hop.vfs.googledrive.GoogleDriveFileSystem.resolveFile(GoogleDriveFileSystem.java:54)
	at org.apache.commons.vfs2.provider.AbstractOriginatingFileProvider.findFile(AbstractOriginatingFileProvider.java:75)
	at org.apache.commons.vfs2.provider.AbstractOriginatingFileProvider.findFile(AbstractOriginatingFileProvider.java:55)
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveFile(DefaultFileSystemManager.java:788)
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveFile(DefaultFileSystemManager.java:754)
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveFile(DefaultFileSystemManager.java:819)
	at org.apache.hop.core.vfs.HopVfs.getFileObject(HopVfs.java:188)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.refreshBrowser(HopVfsFileDialog.java:878)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.enteredFilenameOrFolder(HopVfsFileDialog.java:684)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.lambda$open$2(HopVfsFileDialog.java:329)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4251)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4068)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3645)
	at org.apache.hop.ui.core.vfs.HopVfsFileDialog.open(HopVfsFileDialog.java:581)
	at org.apache.hop.ui.core.dialog.BaseDialog.presentFileDialog(BaseDialog.java:263)
	at org.apache.hop.ui.core.dialog.BaseDialog.presentFileDialog(BaseDialog.java:163)
	at org.apache.hop.pipeline.transforms.fileinput.text.TextFileInputDialog.lambda$open$8(TextFileInputDialog.java:470)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4251)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4068)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3645)
	at org.apache.hop.ui.core.dialog.BaseDialog.defaultShellHandling(BaseDialog.java:513)
	at org.apache.hop.pipeline.transforms.fileinput.text.TextFileInputDialog.open(TextFileInputDialog.java:487)
	at org.apache.hop.ui.hopgui.file.pipeline.delegates.HopGuiPipelineTransformDelegate.editTransform(HopGuiPipelineTransformDelegate.java:174)
	at org.apache.hop.ui.hopgui.file.pipeline.HopGuiPipelineGraph.editTransform(HopGuiPipelineGraph.java:3342)
	at org.apache.hop.ui.hopgui.file.pipeline.HopGuiPipelineGraph.editTransform(HopGuiPipelineGraph.java:2163)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.hop.core.gui.plugin.action.GuiActionLambdaBuilder.lambda$createLambda$0(GuiActionLambdaBuilder.java:96)
	at org.apache.hop.ui.hopgui.context.GuiContextUtil.lambda$handleActionSelection$0(GuiContextUtil.java:183)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4043)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3648)
	at org.apache.hop.ui.hopgui.HopGui.open(HopGui.java:464)
	at org.apache.hop.ui.hopgui.HopGui.main(HopGui.java:351)
Caused by: java.lang.IllegalStateException: You are currently running with version 2.0.1 of google-api-client. You need at least version 1.15 of google-api-client to run version 1.25.0 of the Drive API library.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:534)
	at com.google.api.client.util.Preconditions.checkState(Preconditions.java:113)
	at com.google.api.services.drive.Drive.<clinit>(Drive.java:48)
	... 46 more

--- a/docs/hop-user-manual/modules/ROOT/pages/vfs/google-drive-vfs.adoc
+++ b/docs/hop-user-manual/modules/ROOT/pages/vfs/google-drive-vfs.adoc
@@ -30,7 +30,7 @@ The scheme you can use to access your files in Google Drive is
 == Configuration
 
 You need to generate a credentials file to make it work.
-Follow the Google documentation to see how that is done.
+Follow the https://developers.google.com/drive/api/quickstart/java#authorize_credentials_for_a_desktop_application[Google documentation] to see how that is done.
 You also need to specify a folder in which security tokens are going to be saved.
 You can specify both in the Hop system configuration options.
 This can be done in the Hop GUI: go to the "Google Drive" tab in the Options dialog (from the Tools menu).

--- a/plugins/engines/beam/pom.xml
+++ b/plugins/engines/beam/pom.xml
@@ -170,10 +170,6 @@
                 </exclusion>
                 <exclusion>
                     <groupId>com.google.guava</groupId>
-                    <artifactId>failureaccess</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.google.guava</groupId>
                     <artifactId>guava</artifactId>
                 </exclusion>
                 <exclusion>
@@ -391,10 +387,6 @@
                 <exclusion>
                     <groupId>com.google.errorprone</groupId>
                     <artifactId>error_prone_annotations</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.google.guava</groupId>
-                    <artifactId>failureaccess</artifactId>
                 </exclusion>
                 <exclusion>
                     <groupId>com.google.guava</groupId>
@@ -803,10 +795,6 @@
                 <exclusion>
                     <groupId>com.google.errorprone</groupId>
                     <artifactId>error_prone_annotations</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.google.guava</groupId>
-                    <artifactId>failureaccess</artifactId>
                 </exclusion>
                 <exclusion>
                     <groupId>com.google.guava</groupId>

--- a/plugins/engines/beam/src/assembly/assembly.xml
+++ b/plugins/engines/beam/src/assembly/assembly.xml
@@ -63,6 +63,19 @@
                 <exclude>com.google.auto.value:auto-value-annotations:jar</exclude>
                 <exclude>org.apache.beam:beam-vendor-guava-32_1_2-jre:jar</exclude>
                 <exclude>com.github.luben:zstd-jni:jar</exclude>
+                <exclude>com.google.auth:google-auth-library-oauth2-http:jar</exclude>
+                <exclude>com.google.oauth-client:google-oauth-client:jar</exclude>
+                <exclude>com.google.api-client:google-api-client:jar</exclude>
+                <exclude>com.google.http-client:google-http-client:jar</exclude>
+                <exclude>com.google.http-client:google-http-client-apache-v2:jar</exclude>
+                <exclude>com.google.http-client:google-http-client-appengine:jar</exclude>
+                <exclude>google-http-client-gson:jar</exclude>
+                <exclude>com.google.http-client:google-http-client-jackson2:jar</exclude>
+                <exclude>com.google.android:annotations:jar</exclude>
+                <exclude>org.codehaus.mojo:animal-sniffer-annotations:jar</exclude>
+                <exclude>com.google.api:api-common:jar</exclude>
+                <exclude>com.google.auto.value:auto-value-annotations:jar</exclude>
+                <exclude>com.google.guava:failureaccess:jar</exclude>
             </excludes>
             <outputDirectory>lib/beam</outputDirectory>
         </dependencySet>
@@ -76,6 +89,19 @@
                 <include>com.google.auto.value:auto-value-annotations:jar</include>
                 <include>org.apache.beam:beam-vendor-guava-32_1_2-jre:jar</include>
                 <include>com.github.luben:zstd-jni:jar</include>
+                <include>com.google.auth:google-auth-library-oauth2-http:jar</include>
+                <include>com.google.oauth-client:google-oauth-client:jar</include>
+                <include>com.google.api-client:google-api-client:jar</include>
+                <include>com.google.http-client:google-http-client:jar</include>
+                <include>com.google.http-client:google-http-client-apache-v2:jar</include>
+                <include>com.google.http-client:google-http-client-appengine:jar</include>
+                <include>google-http-client-gson:jar</include>
+                <include>com.google.http-client:google-http-client-jackson2:jar</include>
+                <include>com.google.android:annotations:jar</include>
+                <include>org.codehaus.mojo:animal-sniffer-annotations:jar</include>
+                <include>com.google.api:api-common:jar</include>
+                <include>com.google.auto.value:auto-value-annotations:jar</include>
+                <include>com.google.guava:failureaccess:jar</include>
             </includes>
             <outputDirectory>lib/core</outputDirectory>
         </dependencySet>

--- a/plugins/tech/google/pom.xml
+++ b/plugins/tech/google/pom.xml
@@ -35,13 +35,13 @@
     <properties>
         <commons-vfs2.version>2.9.0</commons-vfs2.version>
         <google-analytics.version>0.65.0</google-analytics.version>
-        <google-api-client.version>2.2.0</google-api-client.version>
-        <google-api-drive.version>v3-rev20230610-2.0.0</google-api-drive.version>
+        <google-api-client.version>2.7.2</google-api-client.version>
+        <google-api-drive.version>v3-rev20250216-2.0.0</google-api-drive.version>
         <google-api-sheets.version>v4-rev20230526-2.0.0</google-api-sheets.version>
         <google-api-sqladmin.version>v1-rev20230721-2.0.0</google-api-sqladmin.version>
         <google-cloud-storage.version>2.15.0</google-cloud-storage.version>
         <google-http-client.version>1.42.3</google-http-client.version>
-        <google-oauth-client.version>1.34.1</google-oauth-client.version>
+        <google-oauth-client.version>1.36.0</google-oauth-client.version>
         <jetty-client.version>6.1.26</jetty-client.version>
     </properties>
 
@@ -79,6 +79,10 @@
                     <groupId>com.google.protobuf</groupId>
                     <artifactId>protobuf-java-util</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.conscrypt</groupId>
+                    <artifactId>conscrypt-openjdk-uber</artifactId>
+                </exclusion>
             </exclusions>
         </dependency>
 
@@ -107,6 +111,12 @@
         <dependency>
             <groupId>com.google.cloud</groupId>
             <artifactId>google-cloud-secretmanager</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.conscrypt</groupId>
+                    <artifactId>conscrypt-openjdk-uber</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
 
         <!-- Google Cloud Storage VFS dependencies -->
@@ -122,6 +132,10 @@
                 <exclusion>
                     <groupId>com.google.protobuf</groupId>
                     <artifactId>protobuf-java-util</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.conscrypt</groupId>
+                    <artifactId>conscrypt-openjdk-uber</artifactId>
                 </exclusion>
             </exclusions>
         </dependency>

--- a/plugins/tech/google/src/assembly/assembly.xml
+++ b/plugins/tech/google/src/assembly/assembly.xml
@@ -47,52 +47,24 @@
         </dependencySet>
         <dependencySet>
             <scope>runtime</scope>
-            <includes>
-                <include>com.google.api:api-common:jar</include>
-                <include>com.google.api-client:google-api-client:jar</include>
-                <include>com.google.api:gax-httpjson:jar</include>
-                <include>com.google.api:gax:jar</include>
-                <include>com.google.api:gax-grpc:jar</include>
-                <include>com.google.api.grpc:proto-google-common-protos:jar</include>
-                <include>com.google.api.grpc:proto-google-iam-v1:jar</include>
-                <include>com.google.apis:google-api-services-analytics:jar</include>
-                <include>com.google.apis:google-api-services-drive:jar</include>
-                <include>com.google.apis:google-api-services-sqladmin:jar</include>
-                <include>com.google.apis:google-api-services-sheets:jar</include>
-                <include>com.google.apis:google-api-services-storage:jar</include>
-                <include>com.google.cloud:google-cloud-core-http:jar</include>
-                <include>com.google.cloud:google-cloud-core:jar</include>
-                <include>com.google.cloud:google-cloud-storage:jar</include>
-                <include>com.google.code.findbugs:jsr305:jar</include>
-                <include>com.google.http-client:google-http-client-appengine:jar</include>
-                <include>com.google.http-client:google-http-client-jackson2:jar</include>
-                <include>com.google.j2objc:j2objc-annotations:jar</include>
-                <include>com.google.oauth-client:google-oauth-client:jar</include>
-                <include>com.google.oauth-client:google-oauth-client-java6:jar</include>
-                <include>com.google.oauth-client:google-oauth-client-jetty:jar</include>
-                <include>io.grpc:grpc-context:jar</include>
-                <include>joda-time:joda-time:jar</include>
-                <include>org.mortbay.jetty:jetty-client:jar</include>
-                <include>org.mortbay.jetty:jetty:jar</include>
-                <include>org.mortbay.jetty:jetty-sslengine:jar</include>
-                <include>org.mortbay.jetty:jetty-util5:jar</include>
-                <include>org.mortbay.jetty:jetty-util:jar</include>
-                <include>org.mortbay.jetty:servlet-api:jar</include>
-                <include>org.threeten:threetenbp:jar</include>
-
-                <include>com.google.analytics:google-analytics-data:jar</include>
-                <include>com.google.api.grpc:proto-google-analytics-data-v1beta:jar</include>
-
-                <include>com.google.cloud:google-cloud-secretmanager:jar</include>
-                <include>com.google.api.grpc:proto-google-cloud-secretmanager-v1beta2:jar</include>
-                <include>com.google.api.grpc:proto-google-iam-v1:jar</include>
-                <include>com.google.api.grpc:proto-google-cloud-secretmanager-v1:jar</include>
-                <include>io.grpc:grpc-inprocess:jar</include>
-                <include>io.grpc:grpc-s2a:jar</include>
-                <include>io.grpc:grpc-util:jar</include>
-                <include>com.google.auth:google-auth-library-oauth2-http:jar</include>
-
-            </includes>
+            <excludes>
+                <exclude>com.google.code.gson:gson</exclude>
+                <exclude>com.google.protobuf:protobuf-java:jar</exclude>
+                <exclude>com.google.protobuf:protobuf-java-util:jar</exclude>
+                <exclude>com.google.auth:google-auth-library-oauth2-http:jar</exclude>
+                <exclude>com.google.oauth-client:google-oauth-client:jar</exclude>
+                <exclude>com.google.api-client:google-api-client:jar</exclude>
+                <exclude>com.google.http-client:google-http-client:jar</exclude>
+                <exclude>com.google.http-client:google-http-client-apache-v2:jar</exclude>
+                <exclude>com.google.http-client:google-http-client-appengine:jar</exclude>
+                <exclude>google-http-client-gson:jar</exclude>
+                <exclude>com.google.http-client:google-http-client-jackson2:jar</exclude>
+                <exclude>com.google.android:annotations:jar</exclude>
+                <exclude>org.codehaus.mojo:animal-sniffer-annotations:jar</exclude>
+                <exclude>com.google.api:api-common:jar</exclude>
+                <exclude>com.google.auto.value:auto-value-annotations:jar</exclude>
+                <exclude>com.google.guava:failureaccess:jar</exclude>
+            </excludes>
             <outputDirectory>plugins/tech/google/lib</outputDirectory>
         </dependencySet>
         <dependencySet>
@@ -100,6 +72,19 @@
                 <include>com.google.code.gson:gson</include>
                 <include>com.google.protobuf:protobuf-java:jar</include>
                 <include>com.google.protobuf:protobuf-java-util:jar</include>
+                <include>com.google.auth:google-auth-library-oauth2-http:jar</include>
+                <include>com.google.oauth-client:google-oauth-client:jar</include>
+                <include>com.google.api-client:google-api-client:jar</include>
+                <include>com.google.http-client:google-http-client:jar</include>
+                <include>com.google.http-client:google-http-client-apache-v2:jar</include>
+                <include>com.google.http-client:google-http-client-appengine:jar</include>
+                <include>google-http-client-gson:jar</include>
+                <include>com.google.http-client:google-http-client-jackson2:jar</include>
+                <include>com.google.android:annotations:jar</include>
+                <include>org.codehaus.mojo:animal-sniffer-annotations:jar</include>
+                <include>com.google.api:api-common:jar</include>
+                <include>com.google.auto.value:auto-value-annotations:jar</include>
+                <include>com.google.guava:failureaccess:jar</include>
             </includes>
             <outputDirectory>lib/core</outputDirectory>
         </dependencySet>

